lib/message_request.js

```diff
@@ -0,0 +1,175 @@
+var logger = require('minilog')('message:request');
+
+var opTable = {
+  control: ['nameSync'],
+  message: ['publish', 'subscribe', 'sync', 'unsubscribe'],
+  presence: ['get', 'set', 'subscribe', 'sync', 'unsubscribe'],
+  status: ['get', 'set', 'subscribe', 'sync', 'unsubscribe'],
+  stream: ['get', 'push', 'subscribe', 'sync', 'unsubscribe']
+};
+
+var Request = function (operation, scope) {
+  this.message = {
+    op: operation,
+    to: scope
+  };
+
+  if (!this._isValid()) {
+    logger.error('invalid request. op: ' + this.op + '; to: ' + this.to);
+  }
+};
+
+Request.buildGet = function (scope, options) {
+  var request = new Request('get', scope);
+  if (request) {
```

Comment:
No, and so no need for these checks after *new* (even current C++ throws an exception when it cannot allocate for *new*)